src/azure-cli/azure/cli/command_modules/storage/__init__.py

```diff
@@ -170,12 +170,14 @@ def register_precondition_options(self):
                    "if it does exist.")
 
     def register_blob_arguments(self):
-        self.extra('blob_name', required=True)
-        self.extra('container_name', required=True)
+        from ._validators import get_not_none_validator
+        self.extra('blob_name', required=True, validator=get_not_none_validator('blob_name'))
+        self.extra('container_name', required=True, validator=get_not_none_validator('container_name'))
         self.extra('timeout', help='Request timeout in seconds. Applies to each call to the service.', type=int)
 
     def register_container_arguments(self):
-        self.extra('container_name', required=True)
+        from ._validators import get_not_none_validator
+        self.extra('container_name', required=True, validator=get_not_none_validator('container_name'))
```

Comment:
Try `az storage blob list -c '""'  --account-name mystorageaccount` in PowerShell, It will pass an empty string and cause a crush.